### ISSUE TITLE
Documentation mismatch: Layer/Model.add_metric() shown usable in call(), but Keras 3 disables / breaks it (TF 2.17 + Keras 3.4)

### ISSUE THREAD

**Issue by griffinstalha** [tensorflow/tensorflow#110884]
### Issue type

Documentation Bug

### Have you reproduced the bug with TensorFlow Nightly?

No

### Source

source

### TensorFlow version

2.17.0

### Custom code

Yes

### OS platform and distribution

Ubuntu 24.04.3 LTS

### Python version

3.10.19

### CUDA/cuDNN version

CUDA: 12.3, cuDNN: 8

### GPU model and memory

4 × NVIDIA RTX 3090 (24GB each)

### Current behavior?

- TensorFlow guide(s) state that layers/models can track metrics via `add_metric()` (often used inside `call()` in subclassed models).
- On TF 2.17.0 + Keras 3.4.1 (TF backend), using `self.add_metric(...)` inside `call()` is not a reliable supported pattern. This aligns with existing reports where `add_metric` is “permanently disabled” in the codepath.
- As a result, the docs example/messaging is misleading for Keras 3 users.

**Expected behavior**
- TensorFlow docs should be updated to not recommend `add_metric()` inside `call()` for Keras 3.
- Docs should show the Keras 3-compatible approach (create a `keras.metrics.Metric` object in `__init__`, call update_state, expose via metrics property, or use a custom `train_step`).

### Standalone code to reproduce the issue

```shell
import os
os.environ["KERAS_BACKEND"] = "tensorflow"

import numpy as np
import tensorflow as tf
import keras

print("TF:", tf.__version__)
print("Keras:", keras.__version__)
print("Backend:", keras.backend.backend())

class M(keras.Model):
    def __init__(self):
        super().__init__()
        self.d = keras.layers.Dense(4)

    def call(self, x):
        y = self.d(x)
        m = keras.losses.MeanSquaredError()(x, y)
        # This is the pattern implied by TF guides ("track metrics via add_metric")
        self.add_metric(m, name="reconstruction_loss")
        return y

x = np.random.randn(8, 4).astype("float32")
m = M()

# Force eager + disable jit to avoid GPU/XLA changing warning surfaces
m.compile(optimizer="adam", loss="mse", run_eagerly=True, jit_compile=False)

try:
    hist = m.fit(x, x, epochs=1, verbose=0)
    print("history keys:", sorted(list(hist.history.keys())))
except Exception as e:
    print("EXCEPTION:", type(e).__name__, str(e))

print("metrics:", [mm.name for mm in m.metrics])
```

### Relevant log output

```shell
2026-02-23 00:24:36.838904: E external/local_xla/xla/stream_executor/cuda/cuda_fft.cc:485] Unable to register cuFFT factory: Attempting to register factory for plugin cuFFT when one has already been registered
2026-02-23 00:24:36.859605: E external/local_xla/xla/stream_executor/cuda/cuda_dnn.cc:8454] Unable to register cuDNN factory: Attempting to register factory for plugin cuDNN when one has already been registered
2026-02-23 00:24:36.865974: E external/local_xla/xla/stream_executor/cuda/cuda_blas.cc:1452] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered
```

**Comment by senderopromail-max** [comment]
<img width="1284" height="488" alt="Image" src="https://github.com/user-attachments/assets/eadd4de7-d1e2-4a41-b220-d98b381008e4" />
![Image](https://github.com/user-attachments/assets/37147fc0-af6b-4e53-b7b7-59ac18ff8be4)
![Image](https://github.com/user-attachments/assets/b4f0c0a7-9637-41ea-bf30-ac5041eff876)
![Image](https://github.com/user-attachments/assets/793b2efa-a3df-40e6-9f35-162e7c845849)
![Image](https://github.com/user-attachments/assets/88fc7dc6-6a87-4c16-9e85-e2927435140d)
![Image](https://github.com/user-attachments/assets/5a240a8d-1eb0-4833-8d11-3050937444ba)
<img width="963" height="1988" alt="Image" src="https://github.com/user-attachments/assets/e85e5619-f533-47bb-9f9f-e0845cecce6c" />
![Image](https://github.com/user-attachments/assets/d67e9a65-8709-412c-a835-73e6a6521a0f)
![Image](https://github.com/user-attachments/assets/e3806fce-e551-4a54-852c-3c3f584ca0d2)
<img width="850" height="500" alt="Image" src="https://github.com/user-attachments/assets/2c5bf1c8-d899-4d4c-8f9d-2500e055d0eb" />
![Image](https://github.com/user-attachments/assets/06441651-cff3-43ee-ae1e-d6aaa466d16f)

**Comment by mvanhorn** [comment]
I've submitted a documentation fix for this in PR #111982. Added a Keras 3 compatibility note to the add_metric() docstring, directing users to the Metric object + update_state() + metrics property approach.

**Comment by nileshau-afk** [comment]
Hi  @griffinstalha ,

Thank you so much for filing this issue. We truly appreciate your support. If you would like to contribute on a fix, please feel free to fork the repository and submit a pull request in accordance with the TensorFlow [CONTRIBUTION GUIDELINES](https://www.tensorflow.org/community/contribute/code) and link this issue in your PR to ensure the maintainers can easily review and collaborate.
To avoid duplicate work, please leave a comment here if you decide to take it on!

we look forward to seeing your contributions!
